angrmanagement/utils/subscriptions.py

```diff
@@ -0,0 +1,58 @@
+import logging
```

Comment:
IMO this is a bad pattern. I set up object containers the way I did in order to minimize the amount of messages that would happen - in that system you explicitly choose when to notify subscribers instead of there being implicit events.

This also just feels like a lot of magic being embedded into places where we really want to know exactly what's going on.